core/Pos/Binary/Crypto.hs

```diff
@@ -50,7 +50,7 @@ instance Bi a => SafeCopy (WithHash a) where
 instance HashAlgorithm algo => Bi (AbstractHash algo a) where
     {-# SPECIALIZE instance Bi (Hash a) #-}
     get = label "AbstractHash" $ do
-        bs <- getByteString $ hashDigestSize @algo $
+        bs <- fmap BS.copy $ getByteString $ hashDigestSize @algo $
```

Comment:
@neongreen @gromakovsky Maybe you can write some comments what's going on here? For others to learn new stuff :)